packages/terra-date-picker/src/DatePickerField.jsx

```diff
@@ -0,0 +1,234 @@
+   */
+  isLabelHidden: PropTypes.bool,
+  /**
+   * An array of ISO 8601 string representation of the dates to enable in the picker. All Other dates will be disabled.
+   */
+  includeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * Custom input attributes to apply to the date input. Use the name prop to set the name for the input.
+   * Do not set the name in inputAttribute as it will be ignored.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  inputAttributes: PropTypes.object,
+  /**
+   * The label of the form control children.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Attributes to attach to the label.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  labelAttrs: PropTypes.object,
+  /**
+   * An ISO 8601 string representation of the maximum date that can be selected.
+   */
+  maxDate: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the minimum date that can be selected.
+   */
+  minDate: PropTypes.string,
+  /**
+   * Name of the date input. The name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   * The first parameter is the event. The second parameter is the changed date value.
+   */
+  onChange: PropTypes.func,
+  /**
+   * A callback function to execute when a change is made in the date input.
+   * The first parameter is the event. The second parameter is the changed date value.
+   */
+  onChangeRaw: PropTypes.func,
+  /**
+   * A callback function to execute when clicking outside of the picker to dismiss it.
+   */
+  onClickOutside: PropTypes.func,
+  /**
+   * A callback function to execute when a date is selected from within the picker.
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Whether or not the date field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * Set the max-width of a field using `length` or `%`.  Best practice recommendation to never exceed
+   * a rendered value of 1020px. _(Note: Providing custom inline styles will take precedence.)_
+   */
+  maxWidth: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the initial value to show in the date input.
+   * This prop name is derived from react-datepicker but is analogous to value in a form input field.
+   */
+  selectedDate: PropTypes.string,
+  /**
+   * Whether or not to append the 'optional' label to a non-required field label.
+   */
+  showOptional: PropTypes.bool,
+};
+
+const defaultProps = {
+  disabled: false,
+  error: null,
+  errorIcon: Field.defaultProps.errorIcon,
+  excludeDates: undefined,
+  filterDate: undefined,
+  help: null,
+  hideRequired: false,
+  includeDates: undefined,
+  inputAttributes: undefined,
+  isInline: false,
+  isInvalid: false,
+  isLabelHidden: false,
+  labelAttrs: {},
+  maxDate: undefined,
```

Comment:
do we need to explicitly set default prop values of undefined?

Reply:
I don't think that any of these default props really need to be set. They were just copied from field and date-picker default props, I was following the pattern of what was done for the other field components.